### DISCUSSION TITLE
How often does the Github app open PRs?

### DISCUSSION THREAD

**Discussion by grug** [renovatebot/renovate#16897] [Request Help] [answered]
Hi there,

I've just configured the Github app with my [test repo](https://github.com/grug/test-renovate) - one of the packages (extract-tld) is out of date but when Renovate runs, it doesn't create a PR updating the package. I was just curious as to how frequently this happens?

Here is the debug log output:

> DEBUG: No dangling containers to remove
INFO: Repository started
{
  "renovateVersion": "32.135.1"
}
DEBUG: Using localDir: /mnt/renovate/gh/grug/test-renovate
DEBUG: PackageFiles.clear() - Package files deleted
{
  "baseBranches": []
}
DEBUG: resetMemCache()
DEBUG: initRepo("grug/test-renovate")
DEBUG: grug/test-renovate default branch = main
DEBUG: Using app token for git init
DEBUG: Repository cache is valid
DEBUG: Resetting npmrc
DEBUG: detectSemanticCommits()
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: Checking cached config file name
DEBUG: Existing config file confirmed
DEBUG: Repository config
{
  "fileName": "renovate.json",
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "config:base"
    ],
    "encrypted": {
      "npmToken": "***********"
    }
  }
}
DEBUG: Repo is onboarded
DEBUG: migrateAndValidate()
DEBUG: No config migration necessary
DEBUG: massaged config
{
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "github>renovatebot/spring-remediations",
      "config:base"
    ],
    "encrypted": {
      "npmToken": "***********"
    }
  }
}
DEBUG: migrated config
{
  "config": {
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "github>renovatebot/spring-remediations",
      "config:base"
    ],
    "encrypted": {
      "npmToken": "***********"
    }
  }
}
DEBUG: Found encrypted config
{
  "config": {
    "npmToken": "***********"
  }
}
DEBUG: Trying to decrypt npmToken
DEBUG: Trying to decrypt with old private key
DEBUG: Decrypted config using default padding
DEBUG: Decrypted npmToken
DEBUG: Migrating npmToken to npmrc
{
  "decryptedToken": "7a********************************48"
}
DEBUG: Adding npmrc to config
DEBUG: Found npmrc in decrypted config - setting
DEBUG: Setting npmrc
DEBUG: Adding token authentication for https://registry.npmjs.org/ to hostRules
DEBUG: Ignoring any .npmrc files in repository due to configured npmrc
DEBUG: Setting hostRules from config
DEBUG: Found repo ignorePaths
{
  "ignorePaths": [
    "**/node_modules/**",
    "**/bower_components/**",
    "**/vendor/**",
    "**/examples/**",
    "**/__tests__/**",
    "**/test/**",
    "**/tests/**",
    "**/__fixtures__/**"
  ]
}
DEBUG: No vulnerability alerts found
DEBUG: No vulnerability alerts found
DEBUG: findIssue(Dependency Dashboard)
DEBUG: Retrieving issueList
DEBUG: Retrieved 2 issues
DEBUG: Found issue 2
DEBUG: No baseBranches
DEBUG: extract()
DEBUG: Found cached extract
{
  "baseBranch": "main",
  "baseBranchSha": "950338c19b67dafe2346b62a49d654f92fc9569f"
}
DEBUG: Deleted cached dep updates
INFO: Dependency extraction complete
{
  "baseBranch": "main",
  "stats": {
    "managers": {
      "npm": {
        "fileCount": 1,
        "depCount": 4
      }
    },
    "total": {
      "fileCount": 1,
      "depCount": 4
    }
  }
}
DEBUG: PackageFiles.add() - Package file saved for branch
{
  "baseBranch": "main"
}
DEBUG: Package releases lookups complete
{
  "baseBranch": "main"
}
DEBUG: branchifyUpgrades
DEBUG: 0 flattened updates found:
DEBUG: Returning 0 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "config": {
    "npm": [
      {
        "packageFile": "package.json",
        "deps": [
          {
            "depType": "dependencies",
            "depName": "@elephant-healthcare/config-provider",
            "currentValue": "^1.5.1",
            "datasource": "npm",
            "prettyDepType": "dependency",
            "depIndex": 0,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/elephant-healthcare/web-packages",
            "currentVersion": "1.5.2",
            "updates": []
          },
          {
            "depType": "dependencies",
            "depName": "extract-tld",
            "currentValue": "^1.0.8",
            "datasource": "npm",
            "prettyDepType": "dependency",
            "depIndex": 1,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/grug/extract-tld",
            "currentVersion": "1.1.0",
            "updates": []
          },
          {
            "depType": "dependencies",
            "depName": "react",
            "currentValue": "^18.2.0",
            "datasource": "npm",
            "prettyDepType": "dependency",
            "depIndex": 2,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/facebook/react",
            "sourceDirectory": "packages/react",
            "homepage": "https://reactjs.org/",
            "currentVersion": "18.2.0",
            "updates": []
          },
          {
            "depType": "dependencies",
            "depName": "typescript",
            "currentValue": "^4.7.4",
            "datasource": "npm",
            "prettyDepType": "dependency",
            "depIndex": 3,
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/Microsoft/TypeScript",
            "homepage": "https://www.typescriptlang.org/",
            "currentVersion": "4.7.4",
            "updates": []
          }
        ],
        "packageJsonName": "test-renovate",
        "packageFileVersion": "1.0.0",
        "packageJsonType": "library",
        "pnpmShrinkwrap": "pnpm-lock.yaml",
        "managerData": {
          "yarnZeroInstall": false,
          "hasPackageManager": false
        },
        "skipInstalls": true,
        "constraints": {},
        "lockFiles": [
          "pnpm-lock.yaml"
        ]
      }
    ]
  }
}
DEBUG: processRepo()
DEBUG: Processing 0 branches:
DEBUG: Calculating hourly PRs remaining
DEBUG: getPrList success
{
  "pullsTotal": 1,
  "requestsTotal": 1,
  "apiQuotaAffected": true
}
DEBUG: currentHourStart=2022-08-01T15:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating prConcurrentLimit (10)
DEBUG: 0 PRs are currently open
DEBUG: PR concurrent limit remaining: 10
DEBUG: Calculated maximum PRs remaining this run
{
  "prsRemaining": 2
}
DEBUG: PullRequests limit = 2
DEBUG: Calculating hourly PRs remaining
DEBUG: currentHourStart=2022-08-01T15:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating branchConcurrentLimit (10)
DEBUG: 0 already existing branches found:
DEBUG: Branch concurrent limit remaining: 10
DEBUG: Calculated maximum branches remaining this run
{
  "branchesRemaining": 2
}
DEBUG: Branches limit = 2
DEBUG: Ensuring Dependency Dashboard
DEBUG: ensureIssue(Dependency Dashboard)
DEBUG: Patching issue
DEBUG: Issue updated
DEBUG: Removing any stale branches
DEBUG: config.repoIsOnboarded=true
DEBUG: No renovate branches found
DEBUG: PackageFiles.clear() - Package files deleted
{
  "baseBranches": [
    "main"
  ]
}
DEBUG: Renovate repository PR statistics
{
  "stats": {
    "total": 0,
    "open": 0,
    "closed": 0,
    "merged": 0
  }
}
DEBUG: Repository result: done, status: onboarded, enabled: true, onboarded: true
DEBUG: Repository timing splits (milliseconds)
{
  "splits": {
    "init": 2640,
    "extract": 378,
    "lookup": 2419,
    "update": 442
  },
  "total": 6671
}
DEBUG: http statistics
{
  "urls": {
    "https://api.github.com/graphql (POST,200)": 2,
    "https://api.github.com/repos/grug/test-renovate/contents/renovate.json (GET,200)": 1,
    "https://api.github.com/repos/grug/test-renovate/issues/2 (GET,200)": 2,
    "https://api.github.com/repos/grug/test-renovate/issues/2 (PATCH,200)": 1,
    "https://api.github.com/repos/grug/test-renovate/pulls (GET,200)": 1,
    "https://api.github.com/repos/renovatebot/spring-remediations/contents/base.json (GET,200)": 1,
    "https://api.github.com/repos/renovatebot/spring-remediations/contents/default.json (GET,200)": 1,
    "https://api.github.com/repos/whitesource/merge-confidence/contents/beta.json (GET,200)": 1,
    "https://registry.npmjs.org/@elephant-healthcare%2Fconfig-provider (GET,200)": 1,
    "https://registry.npmjs.org/extract-tld (GET,200)": 1
  },
  "hostStats": {
    "api.github.com": {
      "requestCount": 10,
      "requestAvgMs": 226,
      "queueAvgMs": 0
    },
    "registry.npmjs.org": {
      "requestCount": 2,
      "requestAvgMs": 1746,
      "queueAvgMs": 0
    }
  },
  "totalRequests": 12
}
INFO: Repository finished
{
  "durationMs": 6671
}

**Comment by viceice** [comment, marked as answer]
https://docs.renovatebot.com/configuration-options/#rangestrategy

**Reply by grug** [reply]
Hi @viceice would you be able to elaborate a little bit more? I'm not seeing anything on that link that is immediately obvious as to what the answer to my question is. I'm not seeing any PRs being created in my repo and I'm trying to get to the bottom of it :)

**Reply by viceice** [reply]
renovate doesn't make PR for inrange update, so it won't update the range until the new version doesn't satisfying the existing range.

**Reply by grug** [reply]
Ah yes that makes sense. Sorry for being silly! I've read through those docs now and this makes sense. Thank you 😄
